resolwe_bio/processes/reads_processing/trimgalore.py

```diff
@@ -0,0 +1,442 @@
+        if inputs.quality_trim.retain_unpaired:
+            params.extend(
+                [
+                    "--retain_unpaired",
+                    "--length_1",
+                    inputs.quality_trim.unpaired_len_1,
+                    "--length_2",
+                    inputs.quality_trim.unpaired_len_2,
+                ]
+            )
+        if inputs.quality_trim.clip_r1:
+            params.extend(["--clip_R1", inputs.quality_trim.clip_r1])
+        if inputs.quality_trim.clip_r2:
+            params.extend(["--clip_R2", inputs.quality_trim.clip_r2])
+        if inputs.quality_trim.three_prime_r1:
+            params.extend(["--three_prime_clip_R1", inputs.quality_trim.three_prime_r1])
+        if inputs.quality_trim.three_prime_r2:
+            params.extend(["--three_prime_clip_R2", inputs.quality_trim.three_prime_r2])
+
+        if inputs.adapter_trim.adapter:
+            for adapter in inputs.adapter_trim.adapter:
+                params.extend(["--adapter", adapter])
+        if inputs.adapter_trim.adapter_2:
+            for adapter in inputs.adapter_trim.adapter_2:
+                params.extend(["--adapter2", adapter])
+        if inputs.adapter_trim.adapter_file_1:
+            params.extend(
+                ["--adapter", f"file:{inputs.adapter_trim.adapter_file_1.fasta.path}"]
+            )
+        if inputs.adapter_trim.adapter_file_2:
+            params.extend(
+                ["--adapter2", f"file:{inputs.adapter_trim.adapter_file_2.fasta.path}"]
+            )
+        if inputs.adapter_trim.universal_adapter:
+            params.append(inputs.adapter_trim.universal_adapter)
+            if (
+                inputs.adapter_trim.adapter
+                or inputs.adapter_trim.adapter_2
+                or inputs.adapter_trim.adapter_file_1
+                or inputs.adapter_trim.adapter_file_2
+            ):
+                self.error(
+                    "You can not supply custom adapter sequence and use "
+                    "the universal adapter sequence."
+                )
+
+        if inputs.hard_trim.trim_5 or inputs.hard_trim.trim_3:
+            self.info(
+                "Only hard trimming was performed. Skipped quality and adapter trimming"
+            )
+        if inputs.hard_trim.trim_5:
+            params.extend(["--hardtrim5", inputs.hard_trim.trim_5])
+        if inputs.hard_trim.trim_3:
+            params.extend(["--hardtrim3", inputs.hard_trim.trim_3])
+
+        return_code, _, _ = Cmd["trim_galore"][params][merged_r1, merged_r2] & TEE(
+            retcode=None
+        )
+        if return_code:
+            self.error("Error while trimming reads.")
+
+        self.progress(0.7)
+
+        trimmed_r1 = f"{name_mate1}_trim.fastq.gz"
+        trimmed_r2 = f"{name_mate2}_trim.fastq.gz"
+        if inputs.hard_trim.trim_5:
+            os.rename(
+                f"{merged_r1[:-9]}.{inputs.hard_trim.trim_5}bp_5prime.fq.gz", trimmed_r1
```

Comment:
`merged_r1[:-9]` and `merged_r2[:-9]` are used several times, it could perhaps be used in a variable.